crates/bcr-ebill-transport/src/handler/mod.rs

```diff
@@ -60,6 +58,14 @@ pub trait BillChainEventProcessorApi: ServiceTraitBounds {
         blocks: Vec<BillBlock>,
         keys: Option<BillKeys>,
     ) -> Result<()>;
+
+    /// Validates that a given bill id is relevant for us, and if so also checks that the sender
+    /// of the event is part of the chain this event is for.
+    async fn validate_chain_event_and_sender(
+        &self,
+        bill_id: &BillId,
+        sender: nostr::PublicKey,
+    ) -> Result<bool>;
```

Comment:
@zupzup with this I was not completely sure. This is to prevent other clients to send us events that change the notification state of a bill. It is actually simple and effective but there is one edge case. If we did not receive the chain invite before the action notification we will reject the message. WDYT?